File: statistics/TypeScript/Median.ts
/**
 * 中位数
 */
import {Rank} from "./rank"
/**
 * 取整
 * @param  {x} x [输入数]
 * @return {~~x}   [返回数字]
 */
function int(x:number):number{
	return ~~x;
}
/**
 * 中位数
 *
 * @param   {Array}  number  需要取中位数的数组
 *
 * @return  {number}         取得的中位数
 */
function Median(number:Array<number>):number{
	var number = Rank.insert(number);
	if(number.length%2==0){
			/**
			 * 当数组长度为偶数，中位数为
			 */
			return number[number.length/2];
	}else{
			/**
			 * 数组长度为奇数,返回
			 * @param {[number]} number[number.length/2]+number[number.length/2+1 [description]
			 */
			return (number[int(number.length/2)]+number[int(number.length/2)+1])/2;
	}
}
export {Median};

File: statistics/python/src/Median.py
# -*- coding: utf-8 -*-
# @Author: Jingyuexing
# @Date:   2020-01-11 17:41:48
# @Last Modified by:   Jingyuexing
# @Last Modified time: 2020-12-03 14:08:31
from Rnak import Rank
def median(data=[]):
    data = Rank.bubbleSort(data)
    if len(data)%2 == 0:
        return data[len(data)/2];
    else:
        return (data[int(len(data)/2)]+data[int(len(data)/2)+1])/2
    pass
